spec/controllers/coupons_controller_spec.rb

```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+describe CouponsController do
+
+  describe "POST create" do
+    context "with valid attributes" do
+      it "creates a new coupon" do
+        expect{
+          post :create, coupon: FactoryGirl.attributes_for(:coupon)
+        }.to change(Coupon,:count).by(1)
+      end
+      
+      it "redirects to the new coupon" do
+        post :create, coupon: FactoryGirl.attributes_for(:coupon)
+        expect(response).to redirect_to Coupon.last
```

Comment:
Did ever cover the fact that if you pass in a model it will auto rails url helper? I don't think we did. If we didn't we need to cover that in a previous reading